doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,679 @@
+   └─ docker logs -f --tail <lines_count> <container_name>
+```
+
+> **Note**
+> I'm using `Docker` as an example fragment extension that provides
+> some `docker` commands. When grouping by `"source"`, we could pull those into
+> a separate top-level entry. When not grouping by `"source"`, those would still
+> show up with the rest of `tasks`. )
+
+#### Store recent commands across sessions
+
+> **Note**
+> _I'm not sure we really want to put this in this spec or not, hence
+> why it is in the "Future considerations" section. I think it is worth
+> mentioning. This might be better served in the [shell integration] doc._
+
+We'll probably want a way for recent commands to be saved across sessions. That way, your `cmd.exe` command history could persist across sessions. We'd need:
+
+* A setting to enable this behavior
+* A setting to control the context of these saved commandlines.
+  * Do we want them saved per-profile, or globally?
+  * If they're saved per-profile, maybe a profile can opt-in to loading all the commands?
+  * How does defterm play with this? Do we "layer: by concatenating per-profile commands with `profiles.defaults` ones?
+* A button in the Settings UI for clearing these commands
+* Should fragments be able to pre-populate "recent commands"?
+  * I'm just gonna say _no_. That would be a better idea for Tasks (aka just a `sendInput` Action that we load from the fragment normally as a Task), or a specific suggestion source for the fragment extension.
+
+#### Inline mode 
+
+> **Note**
+> _This is a half-baked idea with some potential. However, I don't
+> think it needs to be a part of the v1 of the Suggestions UI, so I'm leaving it
+> under future considerations for a future revision._
+
+Do we want to have a suggestions UI "mode", that's just **one** inline
+suggestion, "no" UI? Some UX ala the `PsReadline` recent command suggestion
+feature. Imagine, we just display the IME ghost text thing for the first result,
+given the current prompt? 
+
+Take the following action as an example:
+
+```json
+{ "command": { "action":"suggestions", "source": "commandHistory", "useCommandline": true, "inline": true } },
+```
+
+Type the start of some command at the prompt, and press that key. Presto, we do
+the `pwsh` thing. Ghost text appears for the first match in the `commandHistory`
+for what the user has typed. If they press another key, ~they've typed into the
+"hidden" Suggestions UI, which filters the (hidden) list more, and updates the
+one inline suggestion.~
+
+Or, instead, typed keys go to the shell, and then we re-query the commandline,
+and update the filter accordingly. That would allow tab-completion to still
+work. We'd use <kbd>right arrow</kbd> to accept the suggestion (and dismiss the
+ghost text preview).
+
+This would seemingly SUPER conflict with PowerShell's own handler. Probably not
+something someone should enable for PowerShell 7 profiles if they're using that
+feature. 

```

Comment:
I feel like this would have to be enabled via PowerShell's profile as a KeyHandler for tab completion. That would output an escape sequence that would force WT to open the suggestions UI in inline mode.

Reply:
This is also something _deeper_ shell integration could give us. But then, what is it doing that the shell isn't?